### PR TITLE
[#4408] Calls to code marked @VisibleForTesting in production code

### PR DESCRIPTION
**Pull request by Naresh-kumar-Thodupunoori**
# Pull Request Description

**Title:** feat: Calls to code marked @VisibleForTesting in production code

**Description:**
This pull request addresses an issue where code marked with `@VisibleForTesting` was being called in production code. The `@VisibleForTesting` annotation is typically used to indicate that a method or class should only be accessed from within test code, and its presence in production code can lead to unintended consequences or misuse.

**Changes:**
- Removed the `@VisibleForTesting` annotation from the `loadCatalogBackend` method in `CatalogUtils`.

**Impact:**
- This change ensures that the `loadCatalogBackend` method is clearly marked for use in both production and test environments, aligning with its current usage.

---

By addressing the use of `@VisibleForTesting` annotations in production, this pull request improves code clarity and maintainability.

### CODE REVIEW

**Comment by shaofengshi** [comment]
@Naresh-kumar-Thodupunoori Hello Naresh, could you please create an issue (if not did that), and associate this PR with that issue? so that when the PR got merged, we would know which Issue should be updated. Please follow the PR template as much as possible, thank you!

**Comment by shaofengshi** [comment]
> @Naresh-kumar-Thodupunoori Hello Naresh, could you please create an issue (if not did that), and associate this PR with that issue? so that when the PR got merged, we would know which Issue should be updated. Please follow the PR template as much as possible, thank you!

I see it is #4408 now, please check Justin's comment in the issue.

**Comment by Naresh-kumar-Thodupunoori** [comment]
Hi @shaofengshi could you review my PR

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java [hunk 1]
```diff
@@ -67,7 +67,8 @@ public String login(String keytabFilePath) throws IOException {
     UserGroupInformation.setConfiguration(hadoopConf);
     UserGroupInformation kerberosLoginUgi =
         UserGroupInformation.loginUserFromKeytabAndReturnUGI(catalogPrincipal, keytabFilePath);
-    UserGroupInformation.setLoginUser(kerberosLoginUgi);
+    // Replace setLoginUser with an alternative approach
```
Comment (justinmclean) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: There no real need for this comment in teh code, this wouddl be better placed as a version control comment when making the change.

Comment (Naresh-kumar-Thodupunoori) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: Should I remove the comment and update the PR?
@justinmclean

Comment (justinmclean) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: If you can please that would help

**Comment by Naresh-kumar-Thodupunoori** [comment]
Made the changes accordingly!!

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java (hunk 1 above)
Comment (Naresh-kumar-Thodupunoori) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java]: Made the changes accordingly please review it!!

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public String login(String keytabFilePath) throws IOException {
     UserGroupInformation.setConfiguration(hadoopConf);
     UserGroupInformation kerberosLoginUgi =
         UserGroupInformation.loginUserFromKeytabAndReturnUGI(catalogPrincipal, keytabFilePath);
-    UserGroupInformation.setLoginUser(kerberosLoginUgi);
+    UserGroupInformation.loginUserFromKeytab(catalogPrincipal, keytabFilePath);
```
Comment (jerryshao) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]: @yuqi1129 , please help to review this line.

**Comment by Naresh-kumar-Thodupunoori** [comment]
Hi @jerryshao, @justinmclean, and @shaofengshi,
I hope you're all doing well. I wanted to kindly check in on the status of the PR. As these PRs are connected to my high school grades, I would greatly appreciate any assistance you can offer in expediting the process.
Thank you so much for your understanding and support!

**Comment by justinmclean** [comment]
Hi, again thanks for your help on this, but we need @yuqi1129 to review the change. It shoudl only take a day or two.

Open source is often worked on by volunteers and it is expected that you may not always get a prompt reply, they may be busy with other tasks or only contribite to the project part time.

**Comment by Naresh-kumar-Thodupunoori** [comment]
Yeah I can understand that, but as 15 of august is the last day for my
submissions.I asked you!!

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java (hunk 2 above)
Comment (yuqi1129) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/kerberos/KerberosClient.java:70]:
```suggestion
UserGroupInformation.loginUserFromKeytab(catalogPrincipal, keytabFilePath);
 UserGroupInformation kerberosLoginUgi = UserGroupInformation.getLoginUser();
```

**Comment by Naresh-kumar-Thodupunoori** [comment]
@yuqi1129 Changed the code as per your suggestions!!

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by jerryshao** [comment]
Thanks @Naresh-kumar-Thodupunoori for your contribution, merge to main and branch-0.6.

**Pull request merged**